packages/vite/src/node/plugins/html.ts

```diff
@@ -218,6 +223,61 @@ function handleParseError(
   )
 }
 
+/**
+ * Check if importmap is after any script module. Warn user if so.
+ * Must be added before pre plugins.
+ */
+export function preHtmlImportMapPlugin(config: ResolvedConfig): Plugin {
+  return {
+    name: 'vite:pre-html-importmap',
+    transformIndexHtml: {
```

Comment:
I found that using `transformIndexHtml` does not work for build.

We are running `transformIndexHtml` inside `buildHtmlPlugin` but only user plugins are passed to that.
https://github.com/vitejs/vite/blob/cd69358177dd3d93bc19084ad0ee09f6b85c047c/packages/vite/src/node/plugins/index.ts#L89

So these two plugins won't run for build.

If you change this line to `'head'`, you will see the `postHtmlImportMapPlugin` is not working.
https://github.com/vitejs/vite/pull/9392/files#diff-708ccd9dde483006f589e1440a2af781fc3d08c8f01b517158463e5b075a70abR178

Reply:
Ah thanks for catching this. I thought the tests running in build should've caught this, but looks like my test wasn't accurate. I think this is the case I fear the most as I have to update 

https://github.com/vitejs/vite/blob/4eb159c2c72d7c63a7d24d729e7bbb36f68c36bf/packages/vite/src/node/plugins/html.ts#L840-L860

which feels out of place. But I don't think anywhere else fits.

Reply:
Actually I might extract them out as a hook instead of updating this function, could be a bit cleaner, though boilerplate-y.

Reply:
How about calling `resolveHtmlTransforms` like this? It's a bit odd but maybe it's cleaner.
```js
const [preHooks, postHooks] = resolveHtmlTransforms([preHtmlImportMapPlugin, ...config.plugins, postHtmlImportMapPlugin])

const [preHooks, postHooks] = resolveHtmlTransforms([preHtmlImportMapPlugin, ...server.config.plugins, postHtmlImportMapPlugin])
```

Reply:
I've pushed a commit for this for the `buildHtmlPlugin`. I left `createDevHtmlTransformFn` as is though since it flows better with the `devHtmlHook` in it. (i think)